### PR TITLE
Add Feature::unset_field

### PR DESCRIPTION
**Pull request by lnicola**
- [x] I agree to follow the project's [code of conduct](https://github.com/georust/gdal/blob/master/CODE_OF_CONDUCT.md).
- [x] I added an entry to `CHANGES.md` if knowledge of this change could be valuable to users.
---

### CODE REVIEW

File: src/vector/feature.rs [hunk 1]
```diff
@@ -648,12 +648,28 @@ impl<'a> Feature<'a> {
...
         unsafe { gdal_sys::OGR_F_SetFieldNull(self.c_feature(), idx) };
         Ok(())
     }
 
+    /// Clear a field, marking it as unset.
+    ///
+    /// See: [`OGRFeature::UnsetField`][UnsetField]
+    ///
+    /// [UnsetField]: https://gdal.org/api/ogrfeature_cpp.html#_CPPv4N10OGRFeature10UnsetFieldEi
+    pub fn unset_field(&self, field_name: &str) -> Result<()> {
```
Comment (rouault) [inline src/vector/feature.rs:667]: I guess that should be (same for set_field_null())
```suggestion
    pub fn unset_field(&mut self, field_name: &str) -> Result<()> {
```

Comment (lnicola) [reply src/vector/feature.rs:667]: It should, but all of these currently take `&self` for some reason and @metasim ran into some issues when trying to change them to `&mut self`.

Comment (lnicola) [reply src/vector/feature.rs:667]: That's https://github.com/georust/gdal/issues/499. I think it should work, but didn't try it yet.

**Review by rouault — APPROVED** [review]

**Pull request merged**